cmd/outline-ss-server/main.go

```diff
@@ -346,8 +346,9 @@ func (s *OutlineServer) runConfig(config Config) (func() error, error) {
 								slog.Error("failed to upgrade", "err", err)
 							}
 							defer conn.Close()
-							if clientIP := net.ParseIP(r.RemoteAddr); clientIP != nil {
-								conn = &replaceAddrConn{StreamConn: conn, raddr: &net.TCPAddr{IP: clientIP}}
+							clientAddrPort, err := onet.ParseAddrPortOrIP(r.RemoteAddr)
```

Comment:
I'm not a fan of not knowing our types. This makes our code fragile.
At a minimum, we need to document when to expect an IP and when to expect an addr:port.
But the [http documentation](https://pkg.go.dev/net/http#Request) says this is a host:port. Why do you need to parse as IP?